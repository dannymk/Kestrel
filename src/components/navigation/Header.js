import React from 'react';
import {Navbar} from 'react-bootstrap';

const Header = () => {
   return (
      <Navbar.Header>
         <Navbar.Brand>
            <a href="">{process.env.REACT_APP_name}</a>
         </Navbar.Brand>
      </Navbar.Header>
   );
};

export default Header;