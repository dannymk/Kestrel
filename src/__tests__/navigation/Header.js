import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../components/navigation/Header';

describe("Header portion of menu", () => {
   const title = 'Our title';
   const header = shallow(<Header title={title} />);
   
   test("renders correctly", () => {
      expect(header).toMatchSnapshot();
   });
   
   test("contains the header text", () => {
      expect(header.find('.header-title').text()).toEqual(title);
   });
});