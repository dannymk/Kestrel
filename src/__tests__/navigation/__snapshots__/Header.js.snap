// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header portion of menu renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Our title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <NavbarBrand>
        <a
                className="header-title"
        >
                Our title
        </a>
</NavbarBrand>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <a
          className="header-title"
>
          Our title
</a>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Our title",
          "className": "header-title",
        },
        "ref": null,
        "rendered": "Our title",
        "type": "a",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <NavbarBrand>
          <a
                    className="header-title"
          >
                    Our title
          </a>
</NavbarBrand>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <a
            className="header-title"
>
            Our title
</a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Our title",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "Our title",
          "type": "a",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
